lib/-private/file-path.js

```diff
@@ -0,0 +1,73 @@
+import { parse } from 'node:path';
+
+const DTS = '.d.ts';
+export const GJS = ['.gjs', '.gts'];
+const HBS = '.hbs';
+const HTML = '.html';
+export const SOON_DEPRECATED = ['.js', '.ts'];
+
+function isDTS(ext) {
+  return ext.endsWith(DTS);
+}
+
+function isGlimmerScript(ext) {
+  return GJS.includes(ext);
+}
+
+function isHBS(ext) {
+  return ext === HBS;
+}
+
+function isHTML(ext) {
+  return ext === HTML;
+}
+
+function isScript(ext) {
+  return SOON_DEPRECATED.includes(ext);
+}
+
+export function canProcessFile(filePath) {
+  if (!filePath) {
+    return false;
+  }
+
+  const { ext } = parseFilePath(filePath);
+
+  return isDTS(ext) || isGlimmerScript(ext) || isHBS(ext) || isHTML(ext) || isScript(ext);
+}
+
+export function parseFilePath(filePath) {
```

Comment:
If we're going to parse the file path, let's use the built in utility instead:

https://nodejs.org/api/path.html#pathparsepath

Reply:
I briefly alluded to a problem in https://github.com/ember-template-lint/ember-template-lint/pull/3133#discussion_r1981170408. When a file extension has more than 1 `.`, Node's `parse()` returns just the last part (e.g. `.ts` instead of `.d.ts` or `.css.d.ts`), which makes it hard for us to distinguish files.